### PR TITLE
Add platforms schema command

### PR DESCRIPTION
**Pull request by jquick**
Signed-off-by: Jared Quick <jquick@chef.io>

Fixes https://github.com/inspec/inspec/issues/2488

This requires the downstream train PR to be merged first: https://github.com/inspec/train/pull/347

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,6 +2,8 @@
 source 'https://rubygems.org'
 gemspec name: 'inspec'
 
+gem 'train', :git => 'https://github.com/inspec/train.git', :branch => 'jq/add_platforms_export'
```
Comment (jerryaldrichiii) [inline Gemfile]: I think we can remove this now.

File: inspec.gemspec [hunk 2]
```diff
@@ -26,7 +26,7 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>= 2.3'
 
-  spec.add_dependency 'train', '~> 1.4', '>= 1.4.35'
+  # spec.add_dependency 'train', '~> 1.4', '>= 1.4.35'
```
Comment (jerryaldrichiii) [inline inspec.gemspec]: We should uncomment this :smile:

File: test/functional/inspec_schema_test.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  include FunctionalHelper
+
+  describe 'extracting the platforms schema export' do
+    it 'can execute a platforms schema command' do
+      out = inspec('schema platforms')
+      out.stderr.must_equal ''
+      out.exit_status.must_equal 0
+      data = JSON.parse(out.stdout)
+      data.class.must_equal Array
+      data.size.must_be :>, 1
```
Comment (jerryaldrichiii) [inline test/functional/inspec_schema_test.rb]: Not sure how we officially feel about multiple arguments not being in `()`...but I prefer it.

**Review by jerryaldrichiii — DISMISSED** [review]
Looks great, only a few cleanup items.

File: inspec.gemspec (hunk 2 above)
Comment (jquick) [reply inspec.gemspec]: Yes once train is released we will update these 2

File: Gemfile (hunk 1 above)
Comment (jquick) [reply Gemfile]: Not till train gem is pushed out.

**Review by miah — APPROVED** [review]
Thanks @jquick

**Pull request merged**